src/main/java/com/iota/iri/cache/impl/CacheManagerImpl.java

```diff
@@ -28,7 +30,8 @@ public CacheManagerImpl() {
     public <V> Cache<Indexable, V> getCache(Class<V> type) {
         Cache<Indexable, V> cache = cacheMap.get(type);
         if (cache == null) {
-            cache = add(type);
+            return add(type, new CacheConfigurationImpl(BaseIotaConfig.Defaults.TX_BATCH_WRITE,
```

Comment:
I think you can have `CacheManager` hold an instance of `DBConfig`, then use the normal getter
It will make the code less convoluted (no need for `initializeCaches`) and it may solve the problem maybe?

Reply:
I think `initializeCaches` is clearer. All caches are initialized in 1 place.

Reply:
Maybe move it to `cacheManager` then?
It will be helpful when you turn caching off for benchmarks

Reply:
Moving this to `cacheManager` will also increase complexity by moving DbConfig around the place.
Also, see https://github.com/iotaledger/iri/pull/1706/files#diff-170385153976fbf7aeae0350a52b2894R389-R393 for how cache is turned off.